src/compiler/checker.ts

```diff
@@ -6291,13 +6391,13 @@ namespace ts {
 
         function getTypeOfPropertyOfContextualType(type: Type, name: string) {
             return applyToContextualType(type, t => {
-                let prop = getPropertyOfObjectType(t, name);
+                let prop = t.flags & TypeFlags.StructuredType ? getPropertyOfType(t, name) : undefined;

```

Comment:
Hmm, why did this work before? Is it because union types are flat, and applyToContextualType unwrapped the union types, so all we were left with in lambda was object types? And now unions and intersections can be nested?